front/lib/api/assistant/configuration.ts

```diff
@@ -575,52 +609,49 @@ export async function createAgentConfiguration(
   }
 
   let version = 0;
-  let formerAgentRelationOverride: AgentRelationOverrideType | null = null;
+  let listStatusOverride: AgentUserListStatus | null = null;
 
-  const agentConfig = await front_sequelize.transaction(
+  const agent = await front_sequelize.transaction(
     async (t): Promise<AgentConfiguration> => {
       if (agentConfigurationId) {
-        const [formerAgent] = await AgentConfiguration.findAll({
-          where: {
-            sId: agentConfigurationId,
-            workspaceId: owner.id,
-          },
-          attributes: ["scope", "version"],
-          include: [
-            {
-              model: AgentUserRelation,
-              where: {
-                userId: user.id,
-              },
-              attributes: ["relation"],
-              required: false,
+        const [existing, userRelation] = await Promise.all([
+          AgentConfiguration.findOne({
```

Comment:
Do you think it's worth checking how to keep this to a single query in sequelize?
(since in pure SQL the way to keep it to a single query by adding an outer join is obvious)

Reply:
Same